### PR TITLE
Remove lakectl direct deprecated functionality

### PR DESCRIPTION
**Pull request by nopcoder**
Fix [#6479](https://github.com/treeverse/lakeFS/issues/6479)

### CODE REVIEW

File: esti/adapters_test.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/treeverse/lakefs/pkg/api/helpers"
+
```
Comment (idanovo) [inline esti/adapters_test.go:12]:
```suggestion
```

File: esti/adapters_test.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package helpers
+package esti
```
Comment (idanovo) [inline esti/adapters_test.go:1]: 1. Do we use all the functions defined here only in Esti?
2. I'm not sure about the new file name, it says adapter_test but there are no tests

File: esti/commit_test.go [hunk 3]
```diff
@@ -12,33 +12,25 @@ import (
...
-			getObjResp, err := client.GetObjectWithResponse(ctx, repo, mainBranch, &apigen.GetObjectParams{Path: objPath})
-			require.NoError(t, err, "failed to get object")
-			require.NoErrorf(t, verifyResponse(getObjResp.HTTPResponse, getObjResp.Body),
-				"failed to get object repo %s branch %s path %s", repo, mainBranch, objPath)
-
-			body := string(getObjResp.Body)
-			require.Equalf(t, objContent, body, "path: %s, expected: %s, actual:%s", objPath, objContent, body)
-		})
-	}
+	ctx, _, repo := setupTest(t)
```
Comment (idanovo) [inline esti/commit_test.go:15]: Unrelated question to this PR- how do we check that everything works with pre-signed mode?

File: esti/system_test.go [hunk 4]
```diff
@@ -150,21 +151,81 @@ func uploadFileRandomDataAndReport(ctx context.Context, repo, branch, objPath st
 }
 
 func uploadFileAndReport(ctx context.Context, repo, branch, objPath, objContent string, direct bool) (checksum string, err error) {
+	// Upload using direct access
 	if direct {
-		stats, err := helpers.ClientUploadDirect(ctx, client, repo, branch, objPath, nil, "", strings.NewReader(objContent))
+		stats, err := uploadContentDirect(ctx, client, repo, branch, objPath, nil, "", strings.NewReader(objContent))
```
Comment (idanovo) [inline esti/system_test.go:156]: why you didn't remove the direct flag from the `uploadFileAndReport` and `uploadFileRandomDataAndReport` functions?

File: esti/system_test.go [hunk 5]
```diff
@@ -150,21 +151,81 @@ func uploadFileRandomDataAndReport(ctx context.Context, repo, branch, objPath st
...
+	if err := verifyResponse(resp.HTTPResponse, resp.Body); err != nil {
+		return "", err
+	}
+	return resp.JSON201.Checksum, nil
+}
+
+// uploadContentDirect uploads contents as a file using client-side ("direct") access to underlying storage.
+// It requires credentials both to lakeFS and to underlying storage, but considerably reduces the load on the lakeFS
+// server.
+func uploadContentDirect(ctx context.Context, client apigen.ClientWithResponsesInterface, repoID, branchID, objPath string, metadata map[string]string, contentType string, contents io.ReadSeeker) (*apigen.ObjectStats, error) {
```
Comment (idanovo) [inline esti/system_test.go:176]: I'm not sure I understand why we still need this

File: esti/adapters_test.go (hunk 2 above)
Comment (nopcoder) [reply esti/adapters_test.go:1]: The helper package had adapters that include upload direct.
All the tests that used direct were removed. Except the GC test that I wasn't sure if we like to remove this case - until we will enable different way, like presign.
I've moved the code into the system test (esti) so I've kept the name and added `_test` as this code should be built only when running test.

File: esti/system_test.go (hunk 4 above)
Comment (nopcoder) [reply esti/system_test.go:156]: There are specific GC tests that upload direct so I've moved the implementation and kept the test helper function.

File: esti/system_test.go (hunk 5 above)
Comment (nopcoder) [reply esti/system_test.go:176]: If we do not require direct upload for GC testing, I'll remove this functionality.

**Review by idanovo — APPROVED** [review]
LGTM

File: esti/commit_test.go (hunk 3 above)
Comment (nopcoder) [reply esti/commit_test.go:15]: presign_test.go

Comment (idanovo) [reply esti/commit_test.go:15]: I think that a better way to test it is to run all the tests in another suit with the pre-signed flag on so we don't have to remember to add a new test for every command/feature we add but again, it doesn't related to this PR

**Pull request merged**